src/lib/downloadImage.ts

```diff
@@ -0,0 +1,10 @@
+const downloadImage = (imageUrl: string, fileName: string) => {
```

Comment:
Btw. do you have a reasoning why doing this? 

I understand that components are useful to keep separate, idea is that they should be reusable, there are some nextjs/react optimizations etc. 

But for a function like this... I would probably just keep it in heatmaps.tsx to keep it collocated (not a super strong preference thoguh).

Reply:
I don't have a strong preference. I could see it either way :) I guess I've always pulled these sort of functions out as separate functional units when they have a single role. 🤔  I guess I do it sometimes for reusability, but other times just to group/declutter single discrete tasks out of a larger function/component. 😅 I guess an upside is it would be it's very easy to unit test too, less mocking needed with it having it's own file. 🙃